pkg/gardenlet/controller/shoot/shoot_care_control.go

```diff
@@ -116,26 +116,41 @@ func (c *defaultCareControl) conditionThresholdsToProgressingMapping() map[garde
 	return out
 }
 
-func shootClientInitializer(b *botanistpkg.Botanist) func() error {
+func shootClientInitializer(ctx context.Context, b *botanistpkg.Botanist) func() (bool, error) {
 	var (
-		once sync.Once
-		err  error
+		once             sync.Once
+		apiServerRunning bool
+		err              error
 	)
-	return func() error {
+	return func() (bool, error) {
 		once.Do(func() {
-			err = b.InitializeShootClients()
+			// Don't initialize clients for Shoots, for which the API server is not running
+			apiServerRunning, err = b.IsAPIServerRunning(ctx)
+			if err != nil || !apiServerRunning {
+				return
+			}
+
+			err = b.InitializeShootClients(ctx)
+
+			// double-check if b.K8sShootClient is unset/uninitialized, otherwise we cannot execute
+			// health and constraint checks and garbage collection
+			if b.K8sShootClient == nil {
+				apiServerRunning = false
+			}
```

Comment:
Why is this check needed? Probably you can clarify it in the comment.

Reply:
I added some sentences to the comments. I hope this is more clear now. 
Can you check again?

Reply:
Yep, thanks!